File: src/main/kotlin/engine/objectsData/Loc.kt
package engine.objectsData

import kotlin.math.abs

@kotlinx.serialization.Serializable
data class Loc(val row: Int, val col: Int) {
    fun inRange(loc: Loc, length: Int): Boolean {
        return abs(this.col - loc.col) + abs(this.row - loc.row) <= length
    }
}

File: runner/src/main/resources/apis/ts/src/classes/Loc.ts
export default  class Loc {
    col: number
    row: number

    constructor(col: number, row: number) {
        this.col = col;
        this.row = row;
    }

    public inRange(loc: Loc, range: number) {
        const diff = Math.abs(this.col - loc.col) + Math.abs(this.row - loc.row)
        if (diff === range) return 0
        return diff < range ? 1 : -1
    }
}
